### PR TITLE
Basic Language Menu

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

This is a very _early_ take on a Language menu, making use of the new CharacterItemGroups functionality, as well as functionality added to the trait system that allows it to add and remove languages directly. This PR adds in the languages Tau-Cetic Basic(taking the place of Galactic Common), Tradeband, Freespeak, Elyran Standard, and Sol Common(I'm replacing the extremely basic bastardized Sol Common we had before with the fully realized version of it from Bay12).

<details><summary><h1>Media</h1></summary>
<p>

image

</p>
</details>

# Changelog

:cl:
- add: A basic Languages menu has been added into the Traits tab. 
- add: Tau-Ceti Basic, Tradeband, Freespeak, Elyran Standard, Sol Common, and Sign Language have been added to the Languages tab.

### CODE REVIEW

**Comment by DEATHB4DEFEAT** [comment]
>mental
>>uncatgorized (or some)
>>language

**Comment by VMSolidus** [comment]
> should do >mental >>uncatgorized (or some) >>language

Actually it should probably be a subcategory of Speech. Along with Accents as another subcategory.

**Comment by FoxxoTrystan** [comment]
Please do not change the default GC Languages, this is the more easy for anyone to understand what its is, anything "lore" or dunno related should be choosen by servers not in base EE.

So BY GOD, dod not replace GC, even the Tau-Cetic yml file should NOT have a color if default.

**Comment by FoxxoTrystan** [comment]
Tho characterItemGroup is cool, that i can agree.

**Comment by VMSolidus** [comment]
> Please do not change the default GC Languages, this is the more easy for anyone to understand what its is, anything "lore" or dunno related should be choosen by servers not in base EE.
> 
> So BY GOD, dod not replace GC, even the Tau-Cetic yml file should NOT have a color if default.

No. GC is getting replaced specifically because it's a **Joke** language, and the new one replacing it has actual syllable structures that make it "sound" like real words someone would actually stay instead of "WOOBY GUBB BAZOOP". When I get syllable structures for the remaining languages, they're getting un-joked as well. Having GC in this codebase is kinda unacceptable.

**Comment by VMSolidus** [comment]
image

Having Second-Person text in these is also unacceptable. I went ahead and removed the default color from TCB, and also fixed Sign Language awfullness by removing the genuinely terrible replacements.

File: Resources/Locale/en-US/language/languages.ftl
Comment (FoxxoTrystan) [inline Resources/Locale/en-US/language/languages.ftl:1]: Remove GC from this file.

File: Resources/Prototypes/Language/elyran.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Language/elyran.yml:1]: Please move this file to languages.yml

File: Resources/Prototypes/Language/freespeak.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Language/freespeak.yml:1]: please moves this to languages.ym

File: Resources/Prototypes/Language/languages.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Language/languages.yml:1]: Please fully remove GC here.

Also as there nearly nothing always in language follder, move language.yml to just prototypes folder.

File: Resources/Prototypes/Language/solcommon.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Language/solcommon.yml:1]: move this to languages.yml please

File: Resources/Prototypes/Language/taucetibasic.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Language/taucetibasic.yml:1]: move this to languages.yml please

File: Resources/Prototypes/Language/tradeband.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Language/tradeband.yml:1]: move this to languages.yml please

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
Fine if you wana remove GC, please do it correctly.

Regardning Languages.yml folder, please avoid making multiple variations, just regroup it in one languages.yml file or seperate them like you did but also those in languages.yml

example animals.yml can be their own thing.
Right now slightly, its a tiny mess.

Also is it possible to to edit MaxItems on characterItemGroup with a trait?

File: Resources/Prototypes/Language/elyran.yml
Comment (VMSolidus) [reply Resources/Prototypes/Language/elyran.yml:1]: GOD NO THEY'RE LITERALLY 1000+ LINES.

**Comment by VMSolidus** [comment]
> Fine if you wana remove GC, please do it correctly.
> 
> Regardning Languages.yml folder, please avoid making multiple variations, just regroup it in one languages.yml file or seperate them like you did but also those in languages.yml
> 
> example animals.yml can be their own thing. Right now slightly, its a tiny mess.
> 
> Also is it possible to to edit MaxItems on characterItemGroup with a trait?

Okay! Now that I'm not distracted, I've gone ahead and made improvements to the organization of the Languages folder.
Also, unfortunately I don't think it's currently possible to let a trait modify the max selections of a characterItemGroup. I will actually have to code this, because I've also thought of doing this by the way!

**Review by FoxxoTrystan — APPROVED** [review]

**Comment by VMSolidus** [comment]
Okay, last thing to do. I've gone ahead and setup subcategories for the Speech tab, so that it now includes tabs for Accents and Languages. Which now include more appropriate language related traits. There's also now a characterItemGroup for Accents, so that people can't grief by taking 5+ accents.

image

image

**Review by OldDanceJacket — APPROVED** [review]

**Pull request merged**